src/ERC6538Registry.sol

```diff
@@ -1,21 +1,65 @@
-// SPDX-License-Identifier: UNLICENSED
-pragma solidity 0.8.20;
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.23;
 
 import {IERC6538Registry} from "./interfaces/IERC6538Registry.sol";
 
 /// @dev `ERC6538Registry` contract to map accounts to their stealth meta-address. See
 /// [ERC-6538](https://eips.ethereum.org/EIPS/eip-6538) to learn more.
 contract ERC6538Registry is IERC6538Registry {
-  /// @notice Maps a registrant's identifier to the scheme ID to the stealth meta-address.
+  /// @notice Next nonce expected from `user` to use when signing for `registerKeysOnBehalf`.
   /// @dev `registrant` may be a standard 160-bit address or any other identifier.
   /// @dev `schemeId` is an integer identifier for the stealth address scheme.
-  mapping(bytes registrant => mapping(uint256 schemeId => bytes)) public stealthMetaAddressOf;
+  mapping(address registrant => mapping(uint256 schemeId => bytes stealthMetaAddress)) public
+    stealthMetaAddressOf;
+
+  /// @notice A nonce used to ensure a signature can only be used once.
+  /// @dev `user` is the registrant address.
+  /// @dev `nonce` will be incremented after each valid `registerKeysOnBehalf` call.
+  mapping(address user => uint256 nonce) public nonceOf;
+
+  /// @dev EIP-712 Type hash used in `registerKeysOnBehalf`.
+  bytes32 public constant TYPE_HASH =
+    keccak256("EIP712Domain(string name,string version,uint256 chainId,address registryContract)");
+
+  /// @dev Cache the domain separator as an immutable value.
+  bytes32 private immutable _cachedDomainSeparator;
+  /// @dev Store the chain id that `_cachedDomainSeparator` corresponds to, in order to invalidate
+  /// the cached domain separator if the chain id changes.
+  uint256 private immutable _cachedChainId;
+  /// @dev Cache the address of this contract as an immutable value.
+  address private immutable _cachedThis;
+
+  /// @dev Cache the hash of the name as an immutable value.
+  bytes32 private immutable _hashedName;
+  /// @dev Cache the hash of the version as immutable value.
+  bytes32 private immutable _hashedVersion;
+
+  /// @dev Name of the registry.
+  string private _name;
+  /// @dev Version of the registry.
+  string private _version;
```

Comment:
Whats the point of these if they're only used in the constructor?

Reply:
If aiming to have little diff between OZ code and ours is not a goal, we can remove them. Updated in [b2af4c6](https://github.com/ScopeLift/stealth-address-erc-contracts/pull/5/commits/b2af4c605138ec6f0a8e6361bfa0de28cb140139)